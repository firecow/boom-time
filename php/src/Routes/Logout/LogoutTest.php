<?
declare(strict_types=1);

namespace App\Routes\Logout;

use App\Exceptions\FileErrorException;
use App\Exceptions\FileNotFoundException;
use App\Mocks\ContextMock;
use App\Mocks\SuperGlobalsMock;
use App\Router\RouteHandler;
use App\Testing\RegressionTest;

class LogoutTest extends RegressionTest
{

    /**
     * @return string
     * @throws FileErrorException
     * @throws FileNotFoundException
     */
    public function doTest(): string
    {
        $superGlobals = new SuperGlobalsMock();
        $ctx = new ContextMock($superGlobals);
        $routeHandler = new RouteHandler($ctx);
        $routeResponse = $routeHandler->run("/rest/logout/");
        return $routeResponse->prettyPrint();
    }
}
