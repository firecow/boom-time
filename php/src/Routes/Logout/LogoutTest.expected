HTTP/1.1 200 Ok
Content-Length: 23
Content-Type: text/plain

bearerToken=
expires=Wednesday, 31-Dec-1969 23:59:59 UTC

Logged out successfully